client/app/scripts/liveblog-edit/module.js

```diff
@@ -167,19 +167,52 @@ define([
         });
     }
 
-    BlogSettingsController.$inject = ['blog', 'api'];
-    function BlogSettingsController(blog, api) {
+    BlogSettingsController.$inject = ['$scope', 'blog', 'api', 'blogService', '$location', 'notify'];
+    function BlogSettingsController($scope, blog, api, blogService, $location, notify) {
+        // set view's model
         var vm = this;
-        // set scope
         angular.extend(vm, {
             blog: blog,
-            blogSettings: {},
-            languages: []
+            blogPreferences: angular.copy(blog.blog_preferences),
+            availableLanguages: [],
+            availableThemes: [],
+            isSaved: true,
+            save: function() {
+                // save on backend and clsoe
+                notify.info(gettext('saving blog settings'));
+                blogService.save(vm.blog._id, {blog_preferences: vm.blogPreferences}).then(function(blog) {
+                    vm.isSaved = true;
+                    vm.blog = blog;
+                    notify.pop();
+                    notify.info(gettext('blog settings saved'));
+                    vm.close();
+                });
+            },
+            reset: function() {
+                // reset vm.blogPreferences's values with the ones from global_preferences (backend)
+                api('global_preferences').query().then(function(global_preferences) {
+                    global_preferences._items.forEach(function(item) {
+                        vm.blogPreferences[item.key] = item.value;
+                    });
+                });
+            },
+            close: function() {
+                // return to blog edit page
+                $location.path('/liveblog/edit/' + vm.blog._id);
+            }
         });
-        // load languages
+        // load available languages
         api('languages').query().then(function(data) {
-            vm.languages = data._items;
+            vm.availableLanguages = data._items;
+        });
+        // load available themes
+        api('themes').query().then(function(data) {
+            vm.availableThemes = data._items;
         });
+        // watch if the user selected preferences have changed, in order to update the `isSaved` variable
```

Comment:
I think that we should use a service for these api calls as I should use 3 of them in the exact same way (get for themes, languages and global_preferences)